ffs/scheduler/scheduler_storage.go

```diff
@@ -203,34 +204,47 @@ func (s *Scheduler) StorageConfig(jid ffs.JobID) (ffs.StorageConfig, error) {
 // should be considered failed. If error is nil, it still can return []ffs.DealError
 // since some deals failing isn't necessarily a fatal Job config execution.
 func (s *Scheduler) executeStorage(ctx context.Context, a astore.StorageAction, job ffs.StorageJob, dealUpdates chan deals.StorageDealInfo) (ffs.StorageInfo, []ffs.DealError, error) {
-	ci, err := s.getRefreshedInfo(ctx, a.Cid)
+	ci, err := s.getRefreshedInfo(ctx, a.APIID, a.Cid)
 	if err != nil {
 		return ffs.StorageInfo{}, nil, fmt.Errorf("getting current cid info from store: %s", err)
 	}
 
 	if a.ReplacedCid.Defined() {
-		if err := s.Untrack(a.ReplacedCid); err != nil && err != astore.ErrNotFound {
+		if err := s.Untrack(a.APIID, a.ReplacedCid); err != nil && err != astore.ErrNotFound {
 			return ffs.StorageInfo{}, nil, fmt.Errorf("untracking replaced cid: %s", err)
 		}
 	}
 
-	s.l.Log(ctx, "Ensuring Hot-Storage satisfies the configuration...")
-	hot, err := s.executeHotStorage(ctx, ci, a.Cfg.Hot, a.Cfg.Cold.Filecoin.Addr, a.ReplacedCid)
-	if err != nil {
-		s.l.Log(ctx, "Hot-Storage excution failed.")
-		return ffs.StorageInfo{}, nil, fmt.Errorf("executing hot-storage config: %s", err)
+	var hot ffs.HotInfo
+	if a.Cfg.Hot.Enabled {
+		s.l.Log(ctx, "Executing Hot-Storage configuration...")
+		hot, err = s.executeEnabledHotStorage(ctx, a.APIID, ci, a.Cfg.Hot, a.Cfg.Cold.Filecoin.Addr, a.ReplacedCid)
+		if err != nil {
+			s.l.Log(ctx, "Enabled Hot-Storage excution failed.")
+			return ffs.StorageInfo{}, nil, fmt.Errorf("executing enabled hot-storage: %s", err)
+		}
+		s.l.Log(ctx, "Hot-Storage configuration ran successfully.")
 	}
```

Comment:
If HotStorage is enabled, then we run the hot-storage logic right away which will promote the potentially Staged-pinned Cid (from `pow data stage`) to a full pin.

I say _potentially_ since the Cid might not be staged, and just trusted to be discovered in the IPFS network.

If HotStorage isn't enabled, we run HotStorage logic after ColdStorage logic execution to avoid HotStorage logic unpin the data.